package.json

```diff
@@ -22,17 +22,22 @@
     "debug": "node --inspect ./node_modules/jest-cli/bin/jest --watch --config=./jest.conf.json",
     "format": "prettier --write './src'",
     "lint": "eslint ./src",
-    "precompile": "npm run clean-lib && npm run copy-files && npm run babel",
+    "precompile": "npm run type-check && npm run clean-lib && npm run copy-files && npm run build:types && npm run copy-def-files && npm run babel && npm run generate-proptypes",
     "prepublishOnly": "npm run precompile",
     "postinstall": "node ./scripts/check_carbon_version.js && node ./scripts/check_rfcs.js",
     "watch": "npm run clean-lib && npm run copy-files -- --watch & npm run babel -- --watch",
     "build-storybook": "build-storybook -c .storybook -s .assets",
     "start:static": "npx http-server -p 9001 ./storybook-static",
-    "babel": "cross-env NODE_ENV=production babel ./src --config-file ./babel.config.js --out-dir ./lib --ignore '**/*/__spec__.js','**/*.spec.js','**/__definition__.js','**/*.test.js' --quiet",
+    "babel": "cross-env NODE_ENV=production babel ./src --config-file ./babel.config.js --out-dir ./lib --ignore '**/*/__spec__.js','**/*.spec.js','**/*.spec.ts','**/*.spec.tsx','**/*.test.js','**/*.d.ts' --quiet --extensions '.js','.ts','.tsx'",
     "clean-lib": "rimraf ./lib",
-    "copy-files": "cpy \"**/\" \"!**/(*.js|*.md|*.mdx|*.stories.*|*.snap)\" ../lib/ --cwd=src --parents",
+    "copy-files": "cpy \"**/\" \"!**/(*.js|*.ts|*.tsx|*.md|*.mdx|*.stories.*|*.snap)\" ../lib/ --cwd=src --parents",
     "commit": "git-cz",
-    "generate-metadata": "node ./scripts/generate_metadata/index.mjs"
+    "generate-metadata": "node ./scripts/generate_metadata/index.mjs",
+    "generate-proptypes": "ts-node ./scripts/generate_proptypes/index.ts",
+    "type-check": "tsc --noEmit",
+    "type-check:watch": "npm run type-check -- --watch",
+    "build:types": "tsc -p tsconfig-build.json --emitDeclarationOnly",
+    "copy-def-files": "cpy \"**/(*.d.ts)\" ../lib/ --cwd=src --parents"
```

Comment:
This just copies over the manually defined `.d.ts` files for our js components, it can be removed completely once the whole library in converted to TypeScript.